cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java

```diff
@@ -588,11 +588,16 @@ private ClientTransfer createTransferSync(Transfer transfer)
             HttpPost httpPost = null;
             if (transfer.isQuickTransfer())
             {
-                    Map<String, Object> form = new HashMap<String, Object>();
-                    form.put("TARGET", transfer.getTarget());
-                    form.put("DIRECTION", transfer.getDirection().getValue());
-                    form.put("PROTOCOL", transfer.getProtocols().iterator().next().getUri()); // try first protocol?
-                    httpPost = new HttpPost(vospaceURL, form, false);
+                // Assumption: quick transfer will be a single target
+                if (transfer.targets == null || transfer.targets.size() != 1) {
```

Comment:
If targets is null here or if there are zero targets the error message would be confusing.  Maybe just check `if (transfer.targets.size() > 1`... it wasn't checking for null before.

Reply:
removed the null check (now that transfer.targets is private again.) -